src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -69,15 +72,21 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
 
             try
             {
-                HttpResponse response = await _httpManager.SendGetAsync(_ImdsUri, Headers, logger).ConfigureAwait(false);
+                HttpResponse response = await _httpManager.SendGetAsync(ImdsUri, Headers, logger).ConfigureAwait(false);
+
+                if (response.StatusCode != HttpStatusCode.BadRequest)
+                {
+                    UpdateImdsUriApiVersionAsync(logger);
+                    response = await _httpManager.SendGetAsync(ImdsUri, Headers, logger).ConfigureAwait(false);
+                }
 
                 if (response.StatusCode != HttpStatusCode.OK)
                 {
                     throw new MsalClientException(
-                        MsalError.RegionDiscoveryFailed,
-                        MsalErrorMessage.RegionDiscoveryFailed);
+                    MsalError.RegionDiscoveryFailed,
+                    MsalErrorMessage.RegionDiscoveryFailed);
                 }
-
+                
                 LocalImdsResponse localImdsResponse = JsonHelper.DeserializeFromJson<LocalImdsResponse>(response.Body);
```

Comment:
Possible null ref further down, as `localImdsResponse` can be null.

Reply:
Added a check